frontend/src/types/ProjectTypes.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Possible types of a project
+ * - FSA: Finite state automata
+ * - PDA: Push down automata
+ * - TM: Turing machine
+ */
+export type ProjectType = 'FSA' | 'PDA' | 'TM'
+
+export interface ProjectConfig {
+    acceptanceCriteria: string,
+    // Could be made into enum
+    color: string,
+    statePrefix: string,
+    type: ProjectType
+}
+
+export interface ProjectMetaData {
+    automatariumVersion: string,
+    dateCreated: number,
+    dateEdited: number,
+    name: string,
+    version: string
+}
+
+export interface ProjectComment {
+    x: number,
+    y: number,
+    text: string,
+    id?: number
+}
+
+export interface AutomataState {
+    isFinal: boolean,
+    x: number,
+    y: number,
+    id: number,
+    name?: string,
+    label?: string
+}
+
+export interface AutomataTests {
+    batch: string[],
+    single: string
+}
+
+/**
+ * Possible directions that a TM transitions can move
+ * - Left
+ * - Right
+ * - Stay
+ */
+export type TMDirection = 'L' | 'R' | 'S'
+
+// May be different types for each automata
+export interface AutomataTransition {
```

Comment:
Fine for now, but think we should have different types like the other packages have.
e.g. have a base `Transition` type which has the common stuff and then form other types 
```ts
export interface PDATransition extends AutomataTransition {
  push: string
  pop: string
}
```

Reply:
Yeah for sure 👌